src/main/java/de/retest/surili/commons/test/TestUtil.java

```diff
@@ -0,0 +1,65 @@
+package de.retest.surili.commons.test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import de.retest.recheck.ui.descriptors.Attributes;
+import de.retest.recheck.ui.descriptors.Element;
+import de.retest.recheck.ui.descriptors.IdentifyingAttributes;
+import de.retest.recheck.ui.descriptors.RootElement;
+import de.retest.recheck.ui.descriptors.StringAttribute;
+import de.retest.recheck.ui.descriptors.SutState;
+import de.retest.recheck.ui.image.Screenshot;
+
+public final class TestUtil {
+
+	private TestUtil() {}
+
+	/**
+	 * Creates a new SutState which is equal to another SutState if they contain equal collections of root elements.
+	 *
+	 * @param rootElements
+	 *            The collection of root elements of the SutState.
+	 * @return A new SutState.
+	 */
+	public static SutState createSutState( final RootElement... rootElements ) {
+		return new SutState( Arrays.asList( rootElements ) );
+	}
+
+	public static IdentifyingAttributes getIdentifyingAttributes( final String id ) {
+		return new IdentifyingAttributes( Arrays.asList( new StringAttribute( "a", id ), new StringAttribute( "b", id ),
```

Comment:
The parameter name is misleading. The given `id` is not actually an ID (as described in the Javadoc below), it is only used as a value for the three attributes "a", "b", and "c".

Reply:
It's used as ID since in the comparison of root elements the identifying attributes are compared for equality.  Maybe there is a better way to uniquely define a root element. The child elements are compared as well, so if there is a different number of child elements, the root elements won't be equal.

I used three atrributes since the equality check in recheck looks like this:

```
final IdentifyingAttributes other = (IdentifyingAttributes) object;
return match( other ) > 0.999999999999999;
```

The goal is that only SutStates with root elements with the same ID and number of children will be equal.

Reply:
Ah, I see.

>Maybe there is a better way to uniquely define a root element.

How about the retest ID? But if you need something that is used within `equals`, you have to go with (identifying) attributes. Since `id` is quite misleading, I would rather use something like:

```java
/*
 * @param identifyingAttributesValue
 *            Used as an identifying attributes value to ensure equality as long as the number of children is the same.
 */
```